src/main/java/uk/gov/hmcts/darts/arm/service/impl/UnstructuredToArmProcessorImpl.java

```diff
@@ -0,0 +1,194 @@
+package uk.gov.hmcts.darts.arm.service.impl;
+
+import com.azure.core.util.BinaryData;
+import com.azure.storage.blob.models.BlobStorageException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.darts.arm.api.ArmDataManagementApi;
+import uk.gov.hmcts.darts.arm.config.ArmDataManagementConfiguration;
+import uk.gov.hmcts.darts.arm.service.UnstructuredToArmProcessor;
+import uk.gov.hmcts.darts.common.entity.AnnotationDocumentEntity;
+import uk.gov.hmcts.darts.common.entity.ExternalLocationTypeEntity;
+import uk.gov.hmcts.darts.common.entity.ExternalObjectDirectoryEntity;
+import uk.gov.hmcts.darts.common.entity.MediaEntity;
+import uk.gov.hmcts.darts.common.entity.ObjectDirectoryStatusEntity;
+import uk.gov.hmcts.darts.common.entity.TranscriptionDocumentEntity;
+import uk.gov.hmcts.darts.common.enums.ExternalLocationTypeEnum;
+import uk.gov.hmcts.darts.common.enums.ObjectDirectoryStatusEnum;
+import uk.gov.hmcts.darts.common.enums.SystemUsersEnum;
+import uk.gov.hmcts.darts.common.repository.ExternalLocationTypeRepository;
+import uk.gov.hmcts.darts.common.repository.ExternalObjectDirectoryRepository;
+import uk.gov.hmcts.darts.common.repository.ObjectDirectoryStatusRepository;
+import uk.gov.hmcts.darts.common.repository.UserAccountRepository;
+import uk.gov.hmcts.darts.datamanagement.api.DataManagementApi;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import static uk.gov.hmcts.darts.common.enums.ExternalLocationTypeEnum.ARM;
+import static uk.gov.hmcts.darts.common.enums.ObjectDirectoryStatusEnum.ARM_INGESTION;
+import static uk.gov.hmcts.darts.common.enums.ObjectDirectoryStatusEnum.FAILURE_ARM_INGESTION_FAILED;
+import static uk.gov.hmcts.darts.common.enums.ObjectDirectoryStatusEnum.STORED;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class UnstructuredToArmProcessorImpl implements UnstructuredToArmProcessor {
+
+    private final ExternalObjectDirectoryRepository externalObjectDirectoryRepository;
+    private final ObjectDirectoryStatusRepository objectDirectoryStatusRepository;
+    private final ExternalLocationTypeRepository externalLocationTypeRepository;
+    private final DataManagementApi dataManagementApi;
+    private final ArmDataManagementApi armDataManagementApi;
+    private final UserAccountRepository userAccountRepository;
+    private final ArmDataManagementConfiguration armDataManagementConfiguration;
+
+    @Override
+    @Transactional
+    public void processUnstructuredToArm() {
+        processPendingUnstructured();
+    }
+
+    private void processPendingUnstructured() {
+
+        ObjectDirectoryStatusEntity storedStatus = objectDirectoryStatusRepository.getReferenceById(
```

Comment:
These are constants so would it make sense to initialise these in the constructor so there are less database calls made?